packages/bpk-component-button/src/BpkButtonV2/BpkButton.tsx

```diff
@@ -54,6 +56,28 @@ export const BpkButtonV2 = ({
   const target = blank ? '_blank' : '';
   const rel = blank ? propRel || 'noopener noreferrer' : propRel;
 
+  let processedChildren;
+  if (!disabled && (type === BUTTON_TYPES.link || type === BUTTON_TYPES.linkOnDark)) {
+    processedChildren = Children.map(children, (child) => {
+      if (typeof child === 'string') {
+        const match = child.match(/^(\s*)(.*?)(\s*)$/);
+        const leading = match?.[1] ?? '';
+        const text = match?.[2] ?? '';
```

Comment:
We need consider some edge case to make it more robust, for text part, for example if the children is like `<Icon />Button<Icon /> Link`,  it would underline both text Button and Link, which might not be expected